core/src/main/scala/org/apache/spark/sql/delta/ConflictChecker.scala

```diff
@@ -96,8 +96,9 @@ private[delta] class WinningCommitSummary(val actions: Seq[Action], val commitVe
   val changedDataAddedFiles: Seq[AddFile] = if (isBlindAppendOption.getOrElse(false)) {
     Seq()
   } else {
-    addedFiles
+    addedFiles.filter(_.dataChange)
```

Comment:
this is worth an inline comment explaining why we want this. of course, with the variable name being `changedDataAddedFiles`, this seems obvious. but, obviously, it wasn't :) that's why you are fixing this now :) 

so if you could add an inline comment explaining why this is necessary, e.g. perhaps even with an example, that would be great